### PR TITLE
Enable Clang tidy in CI

### PR DESCRIPTION
**Pull request by clementperon**

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -49,17 +51,32 @@ jobs:
...
+            clang-tidy: ON
 
     steps:
       - name: Checkout code
         uses: actions/checkout@v3
 
-      - name: Configure PcapPlusPlus
+      - if: ${{ matrix.clang-tidy }}
+        name: Configure PcapPlusPlus w/ Clang-Tidy
+        run: CXX=clang++ CC=clang cmake -DLIGHT_PCAPNG_ZSTD=${{ matrix.config-zstd }} -DPCAPPP_ENABLE_CLANG_TIDY=ON -S . -B $BUILD_DIR
```
Comment (seladb) [inline .github/workflows/build_and_test.yml:65]: All we want is to build PcapPlusPlus with `clang-tidy`, no?
In that case, maybe it'd be nicer to add another `job` that just builds PcapPlusPlus without running the tests etc. I think it's a simpler solution than having this `if`... what do you think?

**Review by egecetin — APPROVED** [review]

**Comment by egecetin** [comment]
What is the `cppcheck` version inside alpine container?

**Comment by clementperon** [comment]
> What is the `cppcheck` version inside alpine container?

2.9.1 it's apine package that's why they are new warnings. I will fix them :)
https://pkgs.alpinelinux.org/packages?name=cppcheck&branch=v3.17&repo=&arch=&maintainer=

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -1,28 +1,41 @@
...
-    branches: [ "dev" ]
+    branches: ["dev"]
 
 env:
   BUILD_DIR: Dist
 
 jobs:
   pre-commit:
     runs-on: ubuntu-latest
+    container: seladb/alpine317
```
Comment (seladb) [inline .github/workflows/build_and_test.yml:14]: Why is it needed to run `pre-commit` in a container? 🤔

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -1,28 +1,41 @@
...
+          ./ci/cmake-format-all.sh
+          git diff --exit-code
+
+      - name: Configure PcapPlusPlus for Static analysis
+        run: CXX=clang++ CC=clang cmake -DLIGHT_PCAPNG_ZSTD=ON -DPCAPPP_ENABLE_CLANG_TIDY=ON -S . -B $BUILD_DIR
+
+      - name: Build PcapPlusPlus and check any diff
+        run: |
+          cmake --build $BUILD_DIR -j
+          git diff --exit-code
```
Comment (seladb) [inline .github/workflows/build_and_test.yml:38]: I'm not sure this job should configure and build PcapPlusPlus, we have other jobs that do that. I think this job should be focused on running pre-commit hooks

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (clementperon) [reply .github/workflows/build_and_test.yml:14]: we need cmake, clang, clang-extra-tools

Comment (seladb) [reply .github/workflows/build_and_test.yml:14]: ah ok, that's because we enable `clang-tidy`

**Review by seladb — APPROVED** [review]

**Pull request merged**